utils/common/src/test/java/org/apache/brooklyn/util/HttpAssertsTest.java

```diff
@@ -0,0 +1,293 @@
+    }
+
+    @Test
+    public void shouldAssertHttpStatusCodeEventuallyEquals() throws Exception {
+        stopServer();
+        final String simple = testUri("/simple");
+        HttpAsserts.assertUrlUnreachable(simple);
+        startAfter(2);
+        HttpAsserts.assertHttpStatusCodeEventuallyEquals(simple, 200);
+    }
+
+    private String testUri(String str) {
+        return baseUri.resolve(str).toString();
+    }
+
+    @Test
+    public void shouldAssertContentContainsText() {
+        HttpAsserts.assertContentContainsText(testUri("/simple"), "OK");
+    }
+
+    @Test
+    public void shouldAssertContentNotContainsText() {
+        HttpAsserts.assertContentNotContainsText(testUri("/simple"), "Bad");
+    }
+
+    @Test
+    public void shouldAssertErrorContentsContainsText() {
+        HttpAsserts.assertErrorContentContainsText(testUri("/missing"), "Missing");
+    }
+
+    @Test
+    public void shouldAssertErrorContentNotContainsText() {
+        HttpAsserts.assertErrorContentNotContainsText(testUri("/missing"), "Bogus");
+    }
+
+    @Test
+    public void shouldAssertContentEventuallyContainsText() {
+        stopServer();
+        final String simple = testUri("/simple");
+        HttpAsserts.assertUrlUnreachable(simple);
+        startAfter(2);
+        HttpAsserts.assertContentEventuallyContainsText(simple, "OK");
+    }
+
+    @Test
+    public void shouldAssertContentEventuallyContainsTextWithFlags() {
+        stopServer();
+        final String simple = testUri("/simple");
+        HttpAsserts.assertUrlUnreachable(simple);
+        startAfter(2);
+        HttpAsserts.assertContentEventuallyContainsText(ImmutableMap.of("timeout", "3s"), simple, "OK");
+    }
+
+    @Test(expectedExceptions = AssertionError.class)
+    public void shouldAssertContentEventuallyContainsTextWithTimeout() {
+        stopServer();
+        final String simple = testUri("/simple");
+        HttpAsserts.assertUrlUnreachable(simple);
+        startAfter(4);
+        HttpAsserts.assertContentEventuallyContainsText(ImmutableMap.of("timeout", "3s"), simple, "OK");
+    }
+
+
+    @Test
+    public void shouldAssertContentMatches() {
+        HttpAsserts.assertContentMatches(testUri("/simple"), "[Oo][Kk]");
+    }
+
+    @Test
+    public void shouldAssertContentEventuallyMatches() throws Exception {
+        stopServer();
+        TimeUnit.SECONDS.sleep(2);
+        final String simple = testUri("/simple");
+        HttpAsserts.assertUrlUnreachable(simple);
+        TimeUnit.SECONDS.sleep(2);
+        startAfter(2);
+        HttpAsserts.assertContentEventuallyMatches(testUri("/simple"), "[Oo][Kk]");
+    }
+
+    @Test
+    public void shouldAssertContentEventuallyMatchesWithFlags() {
+        stopServer();
+        final String simple = testUri("/simple");
+        HttpAsserts.assertUrlUnreachable(simple);
+        startAfter(2);
+        HttpAsserts.assertContentEventuallyMatches(ImmutableMap.of("timeout", "3s"), testUri("/simple"), "[Oo][Kk]");
```

Comment:
A couple of example failures:
  * https://builds.apache.org/job/incubator-brooklyn-pull-requests/2069/console
  * https://builds.apache.org/job/incubator-brooklyn-pull-requests/2067/console